File: data-skill/templates/r/explore_variable.R
#!/usr/bin/env Rscript
# Variable Exploration Template
# Purpose: Understand variable types, distributions, and patterns

library(tidyverse)

# ============================================================================
# USER INPUT: Modify these variables
# ============================================================================
data_file <- "data.csv"
variable_name <- "your_variable"  # Change to your variable name
# ============================================================================

# Load data
df <- read_csv(data_file)

cat("\n========================================\n")
cat("VARIABLE EXPLORATION:", variable_name, "\n")
cat("========================================\n\n")

# 1. Check class/type
cat("Variable class:\n")
cat("  ", class(df[[variable_name]]), "\n\n")

# 2. Check unique values
cat("Unique values (n=", n_distinct(df[[variable_name]]), "):\n")
print(table(df[[variable_name]], useNA = "ifany"))

# 3. Check for missing values
missing_count <- sum(is.na(df[[variable_name]]))
missing_pct <- missing_count / nrow(df) * 100
cat("\nMissing values:\n")
cat("  Count: ", missing_count, "\n")
cat("  Percentage: ", sprintf("%.1f%%", missing_pct), "\n\n")

# 4. Check first few values
cat("First 10 values:\n")
print(head(df[[variable_name]], 10))

# 5. Summary statistics (if numeric)
if (is.numeric(df[[variable_name]]) || is.integer(df[[variable_name]])) {
  cat("\nSummary statistics:\n")
  print(summary(df[[variable_name]]))
  cat("\nMean: ", mean(df[[variable_name]], na.rm = TRUE), "\n")
  cat("SD:   ", sd(df[[variable_name]], na.rm = TRUE), "\n")
}

cat("\n========================================\n")

File: data-skill/templates/python/explore_variable.py
#!/usr/bin/env python3
# Variable Exploration Template
# Purpose: Understand variable types, distributions, and patterns

import pandas as pd
import numpy as np

# ============================================================================
# USER INPUT: Modify these variables
# ============================================================================
data_file = "data.csv"
variable_name = "your_variable"  # Change to your variable name
# ============================================================================

# Load data
df = pd.read_csv(data_file)

print("\n" + "=" * 40)
print(f"VARIABLE EXPLORATION: {variable_name}")
print("=" * 40 + "\n")

# 1. Check dtype
print("Variable dtype:")
print(f"  {df[variable_name].dtype}\n")

# 2. Check unique values
print(f"Unique values (n={df[variable_name].nunique()}):")
print(df[variable_name].value_counts(dropna=False))

# 3. Check for missing values
missing_count = df[variable_name].isna().sum()
missing_pct = missing_count / len(df) * 100
print(f"\nMissing values:")
print(f"  Count: {missing_count}")
print(f"  Percentage: {missing_pct:.1f}%\n")

# 4. Check first few values
print("First 10 values:")
print(df[variable_name].head(10))

# 5. Summary statistics (if numeric)
if pd.api.types.is_numeric_dtype(df[variable_name]):
    print("\nSummary statistics:")
    print(df[variable_name].describe())
    print(f"\nMean: {df[variable_name].mean():.2f}")
    print(f"SD:   {df[variable_name].std():.2f}")

print("\n" + "=" * 40)
